app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java

```diff
@@ -70,37 +74,49 @@ public void onMapReady(GoogleMap googleMap) {
 
         applyCustomGoogleMapsStyle();
         populateCampuses();
-        addCampusMarkers();
         addBuildingMarkers();
 
         updateCampus();
     }
 
-    public void switchCampus(boolean loyola){
-    }
-    private void addCampusMarkers() {
-        map.addMarker(LOYOLA_MARKER);
-        map.addMarker(SGW_MARKER);
-    }
 
     private void addBuildingMarkers() {
         ArrayList<Building> sgwBuildings = Campus.SGW.getBuildings();
         ArrayList<Building> loyBuildings = Campus.LOYOLA.getBuildings();
 
+
         for (Building building : sgwBuildings) {
-            map.addPolygon(building.getPolygonOverlayOptions());
-            buildingMarkers.add(map.addMarker(building.getMarkerOptions()));
+            createBuildingMarkersAndPolygonOverlay(building);
         }
 
         for (Building building : loyBuildings) {
-            map.addPolygon(building.getPolygonOverlayOptions());
-            buildingMarkers.add(map.addMarker(building.getMarkerOptions()));
+            createBuildingMarkersAndPolygonOverlay(building);
         }
+        map.setOnPolygonClickListener(new GoogleMap.OnPolygonClickListener() {
+            @Override
+            public void onPolygonClick(Polygon polygon) {
+                ((MainActivity) getActivity()).createToast("Building Clicked");
+            }
+        });
+        map.setOnMarkerClickListener(new GoogleMap.OnMarkerClickListener() {
+            @Override
+            public boolean onMarkerClick(Marker marker) {
+                ((MainActivity) getActivity()).createToast("Building Clicked");
+                return true;
+            }
+        });
+    }
+
+    private void createBuildingMarkersAndPolygonOverlay(Building building) {
+        register(building);
+        map.addPolygon(building.getPolygonOverlayOptions()).setClickable(true);
+
+        Marker marker = map.addMarker(building.getMarkerOptions());
+        building.setMarker(marker);
     }
 
     private void populateCampuses() {
-        Campus.SGW.populateCampusWithBuildings();
-        Campus.LOYOLA.populateCampusWithBuildings();
+        Campus.SGW.populateCampusesWithBuildings();
```

Comment:
I don't populate the campus depending on which one is the "current" one. They both get populated once, when the view is created. However, thank you for pointing this line out. This line currently populate both campuses even thought it seems like only sgw should be populated. I will fix this.